src/main/java/com/google/devtools/build/lib/buildtool/ExecutionTool.java

```diff
@@ -184,6 +187,10 @@ public class ExecutionTool {
         .restrictTo(WorkspaceStatusAction.Context.class, "")
         .restrictTo(SymlinkTreeActionContext.class, "");
 
+    // Register JavaCompileActionContext for java_classpath=bazel to work
+    actionContextRegistryBuilder.register(
+        JavaCompileActionContext.class, new JavaCompileActionContext());
+
```

Comment:
A better location to do this registration is `BazelStrategyModule` overriding `registeringActionContext`. The module allows for better extensibility, that is potential use of a different compilation context.

Please move it there.